openquake/hazardlib/gsim/vanhoutte_2018.py

```diff
@@ -0,0 +1,124 @@
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# OpenQuake is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Module exports :class:`VanHoutteEtAl2018RSD`
+"""
+import numpy as np
+
+from openquake.hazardlib.gsim.base import CoeffsTable, GMPE
+from openquake.hazardlib import const
+from openquake.hazardlib.imt import RSD575
+
+
+class VanHoutteEtAl2018RSD(GMPE):
+    """
+    Implements the GMPE of Van Houtte et al. (2018) for significant duration
+    with 5 - 75 % Arias Intensity. doi:10.1785/0120170076. The oscillator
+    duration model has not yet been implemented.
+    """
+    #: Supported tectonic region type is active shallow crust
+    DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.ACTIVE_SHALLOW_CRUST
+
+    #: Supported intensity measure types are 5 - 75 % Arias
+    #: significant duration
+    DEFINED_FOR_INTENSITY_MEASURE_TYPES = set([RSD575])
+
+    #: Supported intensity measure component is RotD50
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.AVERAGE_HORIZONTAL
+
+    #: Supported standard deviation types are total, inter and intra-event
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = set([
+        const.StdDev.TOTAL,
+        const.StdDev.INTER_EVENT,
+        const.StdDev.INTRA_EVENT
+    ])
+
+    #: Requires vs30
+    REQUIRES_SITES_PARAMETERS = set(('vs30',))
+
+    #: Required rupture parameter is magnitude
+    REQUIRES_RUPTURE_PARAMETERS = set(('mag',))
+
+    #: Required distance measure is closest distance to rupture
+    REQUIRES_DISTANCES = set(('rrup', ))
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+        for spec of input and result values.
+        """
+        C = self.COEFFS[imt]
+        mean = (self.get_magnitude_term(C, rup.mag) +
+                self.get_distance_term(C, dists.rrup, rup.mag) +
+                self.get_site_amplification(C, sites.vs30))
+        print(mean)
+        stddevs = self.get_stddevs(C, dists.rrup.shape, stddev_types)
+        return mean, stddevs
+
+    def get_magnitude_term(self, C, mag):
+        """
+        Returns linear magnitude scaling term
+        """
+        return C["b0"] + C["b1"] * (mag - 6) + C["b2"] * (mag - 6) ** 2
+
+    def get_distance_term(self, C, rrup, mag):
+        """
+        Returns distance scaling term
+        """
+        fac = (rrup > 100).astype('uint8')
```

Comment:
I'm not sure that it is necessary to convert to uint8 here, usually numpy can multiply bools and other vectors.